client/src/+assets/user-profile.tsx

```diff
@@ -0,0 +1,222 @@
+            const credential = await sdk.utils.jwt.generateClientAssertion(accountToAdd)
+            const token = await sdk.marketplace.addNewAddress(credential)
+            localStorage.setItem(StoreItemTypes.MarketplaceApiToken, token)
+            setAddresses([...addresses, newAddress])
+            setNewAddress('')
+            setIsAddressAdded(true)
+            setSuccessMessage('Added is added successfully')
+        } catch (error: any) {
+            if(error.message.includes('"statusCode":401')) {
+                setErrorMessage('Your login is expired. Please change to the previous address, reload and sign again')
+                localStorage.removeItem(StoreItemTypes.MarketplaceApiToken)
+            }
+
+            setErrorMessage(error.message)
+            popupRef.current?.open()
+        }
+        
+    }
+
+    const onSubmitUserProfile = async() => {
+        try {
+            if (!userProfile.nickname) {
+                setInputError('Nickname is required')
+                return
+            }
+            await sdk.profiles.update(userId, userProfile)
+            setIsUpated(true)
+            setSuccessMessage('Your profile is updated successfully')
+            setInputError('')
+        } catch (error: any ) {
+            if(error.message.includes('"statusCode":401')) {
+                newLogin(sdk, loginMarketplaceAPI)
+                setErrorMessage('Your login is expired. Please first sign with your wallet and after try again')
+            } else {
+                setErrorMessage(error.message)
+            }
+            
+            popupRef.current?.open()
+        }
+    }
+
+    useEffect(() => {
+        if(isUpdated || isAddressAdded) {
+            setTimeout(() => {
+                setIsUpated(false)
+                setIsAddressAdded(false)
+                setSuccessMessage('')
+            }, 3000)
+        }
+    }, [isUpdated, isAddressAdded])
+
+    useEffect(() => {
+        (async () => {
+            try {
+                if(!account) {
+                    return
+                }
+                
+                const userProfileData = await sdk.profiles.findOneByAddress(account);
+                setUserId(userProfileData.userId)
+
+                if(userProfileData.addresses.some(a => a.toLowerCase() === newAddress)) {
+                    setNewAddress('')
+                }
+
+                setAddresses([...userProfileData.addresses])
+
+                setUserProfile({
+                    nickname: userProfileData.nickname,
+                    name: userProfileData.name,
+                    email: userProfileData.email,
+                    additionalInformation: userProfileData.additionalInformation,
+                });
+
+                (window as any)?.ethereum?.on('accountsChanged', (accounts: string[]) => {
+                    if(!addresses.some(a => a.toLowerCase() === accounts[0])) {
+                        setNewAddress(accounts[0])
```

Comment:
is the user asked for confirmation in this case?

Reply:
I don't know if I'm understand correctly, this event only is triggered when account changes in the wallet then it check if account is already include in the user profile or not in order to show the input to add new address if it is not already included, I don't think that could be a confirmation process in that moment